UART unavailable on D1 Mini (ESP8226) after ~12hrs

On one of my D1 Minis (ESP8226), after approximately 12 hours of operation (not exactly, but more than a few hours, and less than a day), the unit throws a write error in the CN105 component due to the UART not being connected. This then spams the logs with the below warnings, and no more updates are made or received. This is running the most current branch. A reboot of the D1 solves the problem temporarily. I have not caught it in the act of failing, so there may be an illuminating log message that I'm missing.

My other 4 units do not seem to be having this trouble, so maybe I just have a bad D1 mini, but I wanted to share the log to see if it triggered any ideas.

**Log Excerpt** - the "could not write as asked" line repeats every few seconds, with the checkPendingWantedSettings spamming between.

```
[11:44:49][W][CN105:092]: could not write as asked, because UART is not connected
[11:44:49][W][CN105:097]: delaying packet writing because we need to reconnect first...
[11:44:49][I][EVT_SETS:013]: checkPendingWantedSettings - wanted settings have changed, sending them to the heatpump...
[11:44:49][I][EVT_SETS:013]: checkPendingWantedSettings - wanted settings have changed, sending them to the heatpump...
[11:44:49][I][EVT_SETS:013]: checkPendingWantedSettings - wanted settings have changed, sending them to the heatpump...
[11:44:49][I][EVT_SETS:013]: checkPendingWantedSettings - wanted settings have changed, sending them to the heatpump...
[11:44:49][I][EVT_SETS:013]: checkPendingWantedSettings - wanted settings have changed, sending them to the heatpump...
[11:44:49][I][EVT_SETS:013]: checkPendingWantedSettings - wanted settings have changed, sending them to the heatpump...
[11:44:49][I][EVT_SETS:013]: checkPendingWantedSettings - wanted settings have changed, sending them to the heatpump...
[11:44:49][I][EVT_SETS:013]: checkPendingWantedSettings - wanted settings have changed, sending them to the heatpump...
[11:44:49][I][EVT_SETS:013]: checkPendingWantedSettings - wanted settings have changed, sending them to the heatpump...
```

**UART and CN105 component configuration**

```
uart:
  id: HP_UART
  baud_rate: 2400
  tx_pin: 1
  rx_pin: 3

climate:
  - platform: cn105
    id: hp
    name: "${friendly_name}"
    icon: mdi:heat-pump
    visual:
      min_temperature: 15
      max_temperature: 31
      temperature_step:
        target_temperature: 1.0
        current_temperature: 0.1
#    compressor_frequency_sensor:
#      name: ${name} Compressor Frequency
    vertical_vane_select:
      name: ${name} Vertical Vane
#    horizontal_vane_select:
#      name: ${name} Horizontal Vane
#    isee_sensor:
#      name: ${name} ISEE Sensor
    remote_temperature_timeout: 30min
    update_interval: 4s
    debounce_delay : 500ms
    stage_sensor:
      name: ${name} Stage Sensor
    sub_mode_sensor:
      name: ${name} Sub Mode Sensor
    auto_sub_mode_sensor:
      name: ${name} Auto Sub Mode Sensor
```

jumping in because I have this error

```
In file included from src/esphome.h:31,
                 from src/esphome/components/cn105/Globals.h:2,
                 from src/esphome/components/cn105/cn105.h:2,
                 from src/esphome/components/cn105/hp_readings.cpp:1:
src/esphome/components/cn105/uptime_connection_sensor.h:2:10: fatal error: esphome/components/uptime/uptime_sensor.h: No such file or directory
    2 | #include "esphome/components/uptime/uptime_sensor.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

this is my conf

```
substitutions:
  device_name: little-room-split

esphome:
  name: ${device_name}
  platform: ESP8266
  board: d1_mini

uart:
  id: HP_UART
  baud_rate: 2400
  tx_pin: 1
  rx_pin: 3

wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password
  domain: !secret wifi_domain

  ap:
    ssid: ${device_name} Hotspot
    password: !secret hotspot_password

api:
ota:
web_server:

external_components:
  - source: github://echavet/MitsubishiCN105ESPHome

climate:
  - platform: cn105
    name: Little room split
    update_interval: 10s

logger:
  hardware_uart: UART1
  level: INFO
```

--- a/components/cn105/climate.py
+++ b/components/cn105/climate.py
@@ -20,7 +20,15 @@ from esphome.const import (
 )
 from esphome.core import CORE, coroutine
 
-AUTO_LOAD = ["climate", "sensor", "select", "binary_sensor", "text_sensor", "uart"]
+AUTO_LOAD = [
+    "climate",
+    "sensor",
+    "select",
+    "binary_sensor",
+    "text_sensor",
+    "uart",
+    "uptime",
+]
 
 CONF_SUPPORTS = "supports"
 # from https://github.com/wrouesnel/esphome-mitsubishiheatpump/blob/master/components/mitsubishi_heatpump/climate.py

